src/main/java/com/telefonica/iot/tidoop/backends/ckan/CKANBackend.java

```diff
@@ -0,0 +1,270 @@
+ * francisco.romerobueno at telefonica dot com
+ */
+
+package com.telefonica.iot.tidoop.backends.ckan;
+
+import com.telefonica.iot.tidoop.http.HttpClientFactory;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.log4j.Logger;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+
+/**
+ *
+ * @author frb
+ */
+public class CKANBackend {
+    
+    private final Logger logger;
+    private final String ckanHost;
+    private final String ckanPort;
+    private final String apiKey;
+    private final boolean ssl;
+    private HttpClientFactory httpClientFactory;
+    
+    /**
+     * Constructor.
+     * @param ckanHost
+     * @param ckanPort
+     * @param ssl
+     * @param apiKey
+     */
+    public CKANBackend(String ckanHost, String ckanPort, boolean ssl, String apiKey) {
+        this.ckanHost = ckanHost;
+        this.ckanPort = ckanPort;
+        this.logger = Logger.getLogger(CKANBackend.class);
+        this.apiKey = apiKey;
+        this.ssl = ssl;
+        this.httpClientFactory = new HttpClientFactory(ssl);
+    } // CKANBackend
+    
+    /**
+     * Sets the Http client factory. It is protected since it is only going to be used by the tests.
+     * @param httpClientFactory
+     */
+    protected void setHttpClientFactory(HttpClientFactory httpClientFactory) {
+        this.httpClientFactory = httpClientFactory;
+    } // setHttpClientFactory
+    
+    /**
+     * Gets the package identifiers/names within a given organization id/name.
+     * @param orgId
+     * @return The package identifiers/names within the given organization id/name
+     */
+    public List<String> getPackages(String orgId) {
+        logger.debug("Getting the packages within " + orgId + " organization");
+
+        try {
+            List<String> resIds = new ArrayList<String>();
+            String url = "http" + (ssl ? "s" : "") + "://" + ckanHost + ":" + ckanPort
+                    + "/api/3/action/organization_show?id=" + orgId;
+            CKANResponse resp = doCKANRequest("GET", url, "");
+            JSONObject result = (JSONObject) resp.getJsonObject().get("result");
+            JSONArray pkgs = (JSONArray) result.get("packages");
+            
+            for (Object pkg1 : pkgs) {
+                JSONObject pkg = (JSONObject) pkg1;
+                resIds.add((String) pkg.get("id"));
+            } // for
+            
+            return resIds;
+        } catch (Exception e) {
+            return null;
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-tidoop/commit/6192985f02fa37890506e4a3cd12fcea35364cbd